yt_dlp/extractor/yandexvideo.py

```diff
@@ -149,16 +150,22 @@ def _real_extract(self, url):
 
 
 class YandexVideoPreviewIE(InfoExtractor):
-    _VALID_URL = r'''(?x)
-                    https?://
-                        (?:
-                            yandex\.ru/video/preview(?:/?\?.*?filmId=|/)
-                        )
-                        (?P<id>\d+)
-                    '''
+    _VALID_URL = r'https?://(?:www\.)?yandex\.ru/video/preview/?(?:\?.*?filmId=|)(?P<id>\d+)'
```

Comment:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?yandex\.ru/video/preview(?:/?\?.*?filmId=|/)(?P<id>\d+)'
```
This is more correct, no?